pkg/pillar/cmd/downloader/datastoreconfig.go

```diff
@@ -35,3 +36,22 @@ func handleDatastoreConfigDelete(ctxArg interface{}, key string,
 	ctx.pubCipherBlockStatus.Unpublish(cipherBlock.Key())
 	log.Functionf("handleDatastoreConfigDelete for %s", key)
 }
+
+//notifyResolveConfigByNewDatastoreConfig fires modify handler for ResolveConfig
+//we need to call it in case of no DatastoreConfig found
+func notifyResolveConfigByNewDatastoreConfig(ctxArg interface{}, key string,
```

Comment:
This function should be called with the ctx and config already type cast.

Reply:
I might be even easier to understand if you move the code from checkAndUpdateDownloadableObjects here. Then the function will iterate over the two types of objects and call the modify handler.

Note that I don't think it matters whether you interate over *Config as here or over *Status as in checkAndUpdateDownloadableObjects, so I'd make that be the same as well.